src/index.js

```diff
@@ -1 +1,275 @@
+        </form>
+    `;
+    modalDetail.innerHTML = html;
+    modalDetail.classList.remove('hide');
+    modalDetail.classList.add('show');
+
+    // Add the comments
+    const commentsSection = document.getElementById('user-comments');
+    const userName = document.querySelector('form .name-input');
+    const userComment = document.querySelector('form .enter-comment');
+    const form = document.querySelector('.comments-form');
+    const commentsHeader = document.querySelector('.comments-header');
+
+    // Create new entry when user submits a new comment
+
+    const listComment = async () => {
+      let scoreArray = [];
+      const addToList = async () => {
+        const result = await fetch(
+          `https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/bbDC3TOidzHVfwfLZkFs/comments?item_id=${foodID}`,
+        ).then((res) => res.json());
+        return result;
+      };
+      addToList().then((res) => {
+        commentsSection.innerHTML = '';
+        if (res) {
+          scoreArray = res;
+          for (let i = 0; i < scoreArray.length; i += 1) {
+            commentsSection.innerHTML += `
+                  <li> ${scoreArray[i].creation_date}: ${scoreArray[i].username}-${scoreArray[i].comment}</li>
+                    `;
+          }
+        }
+      });
+      const wait = await addToList();
+      const data = scoreArray.length;
+      commentsHeader.innerHTML = ` (${data}) Comments `;
+    };
+    listComment();
+
+    form.addEventListener('submit', async (e) => {
+      e.preventDefault();
+      postComment(foodID, userName.value, userComment.value); // via fetch api
+      form.reset();
+    });
+  }
+};
+
+// p/s: The display should not be ties to the submit event listener.
+
+mealList.addEventListener('click', getRecipe);
+
+// Add event listener to the close button
+modalDetail.addEventListener('click', (e) => {
+  if (e.target.classList.contains('fa-xmark')) {
+    const modal = e.target.parentElement.parentElement;
+    modal.classList.remove('show');
+    modal.classList.add('hide');
+  }
+});
+
+// Involvement API to track the likes
+// bbDC3TOidzHVfwfLZkFs
+
+// Create the add like functionality
+mealList.addEventListener('click', async (e) => {
+  if (e.target.classList.contains('fa-heart')) {
+    const mainList = e.target.parentElement.parentElement.parentElement.parentElement
+      .parentElement;
+    const id = mainList.getAttribute('meal-id');
+    postLike(id);
+    const updateLikes = mainList.lastElementChild.children[1];
+    likesData = await renderLike();
+
+    likesData.forEach((entry) => {
+      if (entry.item_id === id) {
+        updateLikes.innerHTML = `${entry.likes} Likes`;
+      }
+    });
+  }
+});
+
+const postLike = async (mealId) => {
+  const like = {
+    item_id: mealId,
+  };
+
+  const response = await fetch(
+    'https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/bbDC3TOidzHVfwfLZkFs/likes',
```

Comment:
The project uses the Involvement API to record the different user interactions (likes and comments) 📝